lib/src/main/kotlin/com/swmansion/starknet/provider/rpc/JsonRpcProvider.kt

```diff
@@ -68,6 +66,28 @@ class JsonRpcProvider(
         return HttpRequest(payload, buildJsonHttpDeserializer(responseSerializer, deserializationJson), httpService)
     }
 
+    fun <T> combineBatchRequest(
+        calls: List<Request<T>>,
+    ): BatchHttpRequest<T> {
+        val httpRequests = calls.map { it as HttpRequest<T> }
```

Comment:
To be a bit more type safe, maybe we could first verify the elements are indeed HttpRequest<T>, or at least handle the possible exception?

Reply:
Correct, I've changed the type from `Request` to `HttpRequest` 👍